app/uk/gov/hmrc/gform/des/DesConnector.scala

```diff
@@ -53,12 +54,18 @@ class DesConnector(wSHttp: WSHttp, baseUrl: String, desConfig: DesConnectorConfi
       authorization = Some(Authorization(desConfig.authorizationToken)))
     Logger.info(
       s"Des lookup, Tax Periods: '$idType, $idNumber, $regimeType', ${loggingHelpers.cleanHeaderCarrierHeader(hc)}")
-    wSHttp.GET[Obligation](
-      s"$baseUrl${desConfig.basePath}/enterprise/obligation-data/$idType/$idNumber/$regimeType?status=O") recoverWith {
-      case e: JsValidationException => Future(Obligation(List()))
+    val value = s"$baseUrl${desConfig.basePath}/enterprise/obligation-data/$idType/$idNumber/$regimeType?status=O"
+    val response = wSHttp.GET[JsValue](value)
+    val checkObligation = response.map(i => i.validate[Obligation])
+    val checkNoPeriods = response.map(i => i.validate[NoPeriods])
+    if (checkObligation.isInstanceOf[Future[JsSuccess[Obligation]]]) {
```

Comment:
You can't do these isInstances of here, because check* are Futures

Instead should map over response, so they are not Futures. Then within that you can almost certainly match with cases: if else if else is an anti-pattern here